File: Silver/1000-1999/1000-1099/1072/1072.go
package main

import "fmt"

func Silver1072() {
	var x, y int
	fmt.Scan(&x, &y)

	result := 0
	// z := (y / x) * 100
	z := (y * 100) / x // 이렇게 해야 소수점이 버려지지 않음

	if z >= 99 || x == y {
		fmt.Println(-1)
		return
	}

	left, right := 0, 1000000000
	for left <= right {
		mid := (left + right) / 2
		curr_z := ((y + mid) * 100) / (x + mid)

		if z < curr_z {
			right = mid - 1
			result = mid
		} else {
			left = mid + 1
		}
	}

	fmt.Println(result)
}

File: Silver/1000-1999/1000-1099/1072/1072.py
import sys 
input = sys.stdin.readline

def solution(X,Y):
    z = (Y * 100) // X
    start, end = 0, 1_000_000_000
    
    result = 0
    while start <= end:
        mid = (start + end)//2
        curr_z = ((Y+mid)*100)//(X+mid)
        # print(curr_z)
        
        if z < curr_z:
            end = mid -1 
            result = mid
        else:
            start = mid + 1
        
    if result == 0:
        return -1

    return result 
        
        
if __name__ == '__main__':
    X, Y = map(int,input().split())
    print(solution(X,Y))